src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.cs

```diff
@@ -165,16 +174,14 @@ public void WriteEntry(TarEntry entry)
         {
             ThrowIfDisposed();
 
-            TarHelpers.VerifyEntryTypeIsSupported(entry.EntryType, Format, forWriting: true);
-
             WriteGlobalExtendedAttributesEntryIfNeeded();
 
             byte[] rented = ArrayPool<byte>.Shared.Rent(minimumLength: TarHelpers.RecordSize);
             Span<byte> buffer = rented.AsSpan(0, TarHelpers.RecordSize); // minimumLength means the array could've been larger
             buffer.Clear(); // Rented arrays aren't clean
             try
             {
-                switch (Format)
+                switch (entry.Format)
```

Comment:
Why do we have a "default" Format for a TarWriter, if we are just going to throw an exception when the TarEntryFormat is "Unknown"?

Reply:
This is not the method that will consume the default format. This is the method that will write the entry using the format of the passed `entry`.

Reply:
Are you saying this `WriteEntry` method should default to the `default` Format is an entry happens to have `Unknown` as its underlying format?

Reply:
I see. I thought I was going to get a warning saying that there was no `default` in the switch. But it seems that only happens when you don't add all the cases.